programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -1628,10 +1572,11 @@ declare_builtin_function!(
                     })
                     .collect::<Result<Vec<_>, InstructionError>>()?;
                 accounts.clone_from_slice(account_metas.as_slice());
+            } else {
+                result_header.data_len = instruction_context.get_instruction_data().len() as u64;
+                result_header.accounts_len =
+                    instruction_context.get_number_of_instruction_accounts() as u64;
             }
-            result_header.data_len = instruction_context.get_instruction_data().len() as u64;
-            result_header.accounts_len =
-                instruction_context.get_number_of_instruction_accounts() as u64;
```

Comment:
Why is this in the else branch now?

Reply:
To ensure that we don't use an invalidated `result_header` when the "then" branch of the `if` was taken.